File: Resources/MPI_Hello_World-20240227/program_hello_mpi.c
#include<stdio.h>
#include<string.h>
#include<mpi.h> 		/* new and contains the function prototypes, macros, variables.. */

int main(int argc, char** argv)
{
  int i, myid, size, tag=100;
  char message_send[50], message_recv[50];
  MPI_Status status; 		/* data type that is defined in mpi.h... */

  MPI_Init(&argc, &argv);
  
  MPI_Comm_size(MPI_COMM_WORLD, &size); /* tells about the number of processes */
  MPI_Comm_rank(MPI_COMM_WORLD, &myid); /* will return the rank or id of the process that called it.  */

  /* printf("%d", size); */

  if (myid != 0) 		/* not the designated process to receive..send the data.. */
    {
      sprintf(message_send, " Hello from process id: %d \n ", myid);
      MPI_Send(message_send, 50, MPI_CHAR, 0, tag, MPI_COMM_WORLD);
    }
  else
    {				/* only by process id = 0, myrank = 0, myid = 0 */
      for (i = 1; i < size; i++)
	{
	  MPI_Recv(message_recv, 50, MPI_CHAR, i, tag, MPI_COMM_WORLD, &status);
	  printf("\n %s", message_recv); 
	}

      sprintf(message_send, " Hello from process id: %d \n ", myid);
      printf("\n %s", message_send); 

    }

  MPI_Finalize();

  return 0;
}

File: Resources/MPI_Hello_World-20240227/program_hello_mpi.f90
program main
  implicit none
  include "mpif.h"
  integer :: i, myid, size, mpierror, tag, status(MPI_STATUS_SIZE)
  character(len=50) :: message_send, message_recv

  call MPI_INIT(mpierror)

  call MPI_COMM_SIZE(MPI_COMM_WORLD, size, mpierror)
  call MPI_COMM_RANK(MPI_COMM_WORLD, myid, mpierror)

  tag = 100

  if (myid /= 0) then 
     write(message_send, *), 'Hello from process id:', myid
     call MPI_SEND(message_send, 50, MPI_CHARACTER, 0, tag, MPI_COMM_WORLD, mpierror)
  else
     do i = 1, size-1
        call MPI_RECV(message_recv, 50, MPI_CHARACTER, i, tag, MPI_COMM_WORLD, status, mpierror)
        write(*, *), message_recv
     end do
     write(message_send, *), 'Hello from process id:', myid     
     write(*, *), message_send
  end if

  call MPI_FINALIZE(mpierror)

end program main
